### ISSUE TITLE
Dec 14 - Use Machine Learning to Predict Rocket Launch Delays

### ISSUE THREAD

**Issue by meaghanlewis** [microsoft/ReactorPlanning#125]
# Event Information
| Category | Details |
|-----------|---------|
| Reactor Topic |  Data Science and Machine Learning | 
| Workshop Duration | 60 min | 
| Title | Use Machine Learning to Predict Launch Delays |
| Description | This workshop introduces you to the world of machine learning. You'll take a real-life problem that NASA faces and apply machine learning to solve it. The goal is to discover how machine learning could help solve other problems in space discovery and different aspects of life. This workshop begins with an introduction to rocket launches, and then moves on to learn how to import and clean data for use in creating machine learning models. During this workshop, we'll be going through this content on Microsoft Learn: https://aka.ms/RocketDelayswithML-2. To join this workshop, you should have explored the role of Python in space exploration path on Microsoft Learn: https://aka.ms/PythoninSpace-2 and have Visual Studio Code and Python installed. You should also have the ability to code simple programs in Python. |
| Content| https://aka.ms/RocketDelayswithML-8 modules 1-2 | 

# Workshop(s) details
| Date |  Local Start Time |  Location |  Speaker | SharePoint ID | Platform |
|------|-------------------|-----------|----------|---------------|----------|
| 12/15 | 8:30 PM | Bangalore | Michael Galarnyk | 11894 |----------|
| 12/15 | 4:00 PM | US, Canada, Sydney | Michael Galarnyk | 11895 |----------|
| 12/16 | 7:30 PM | Sao Paulo | Flavio Figueiredo | 11896 |----------|
| 12/16 | 7:30 PM | Shanghai | Ryan Chung | 11897 |----------|
| 12/16 | 12:00 PM | London, Stockholm, Abu Dhabi, Tel Aviv | Brian Sletten | 11898 |----------|

# Event Checklist

## 10 Weeks Prior
- [x] Reactor Content Team (RCT) opens a new issue for an upcoming local workshop filling in the Event Information and Workshop details tables. RCT member who owns content (content owner) this week will be the assignee.
- [x] RCT and content owner discuss details of the event in the comments of this issue. RCT adds a _confirmed-event_ label and _local-digital-event_ tag to the issue and adds event to the __Local Digital Events__ project. Issue will appear in the __Upcoming Streams__ column
- [x] RCT adds a _ready-for-marketing-edit_ label so the RCT can review title and description

## 8 Weeks Prior
- [ ] RCT does a review of title and description and removes the _ready-for-marketing-edit_ label 
- [ ] RCT moves the issue to the __Ready for SharePoint__ column
- [ ] RCT informs the Reactor Marketing Team of the new event information
- [ ] RCT confirms the speaker
- [ ] RCT provides speaker with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT moves the issue to the __Content Preparation In Progress__ column

## 5 weeks Prior
- [ ] Content owner opens a PR with the new content for the workshop on the [customer-facing Reactor repository](https://github.com/microsoft/Reactors/tree/main/online-event-resources) and adds the _content-development_ and _local-digital-event_ tag to the PR. 
- [ ] RCT reviews, approves, and merges the changes
- [ ] RCT shares content with the vendor/instructor
- [ ] RCT provides vendor/instructor with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT ensures that instructors have received an invite to the workshops and that PM team works with instructors to get setup in the platform (e.g. Teams, Teams Live Event)

## 1 Week Prior
- [ ] Connect vendor/instructor with the PM(s) for the locations where this event has been marketed
- [ ] Check-in with vendor/instructor that they are prepared for the workshop
- [ ] RCT moves the issue to the __Ready to Stream__ column

## During and After Stream
- [ ] Instructor (and optional moderator) run the stream
- [ ] _(Optional)_RCT submits recording to be uploaded to the Reactor YouTube and to be made available on LearnTV
- [ ] RCT moves issue to the __Stream Complete__ column

**Comment by mebaumb** [comment]
This looks good to me - 

This workshop introduces you to the world of machine learning. You'll take a real-life problem that NASA faces and apply machine learning to solve it. The goal is to discover how machine learning could help solve other problems in space discovery and different aspects of life. This workshop begins with an introduction to rocket launches, and then moves on to learn how to import and clean data for use in creating machine learning models. During this workshop, we'll be going through this content on Microsoft Learn: https://aka.ms/RocketDelayswithML-2

To join this workshop, you should have explored the role of Python in space exploration path on Microsoft Learn: https://aka.ms/PythoninSpace-2 and have Visual Studio Code and Python installed. You should also have the ability to code simple programs in Python.

**Comment by meaghanlewis** [comment]
@mebaumb thanks! We're going through modules 1 and 2 of this learn path: https://aka.ms/RocketDelayswithML-8, which recommends: https://aka.ms/SpacewithPython-8 as a pre-req.

Once we have the other link generated we can add that in as what will be covered.

**Comment by meaghanlewis** [comment]
Here's an overview of what this workshop will cover and suggested agenda to follow with links and resources.

## Content to teach
This workshop will take Reactor community members through modules **1 and 2** of the Microsoft Learn path: [Use machine learning to predict rocket launch delays](https://aka.ms/RocketDelayswithML-8)

**SHARE THIS LINK WITH REACTOR ATTENDEES: https://aka.ms/RocketDelayswithML-8**

**Encourage attendees to go through the modules with you and complete it. Also encourage them to rate the module at the end.**

## Agenda
1. **Intro/Overview** 5 min
   - Welcome community members
   - Provide overview of content to be covered
   - Show the learning path
   - Encourage community members to complete and rate the modules as they go along
   - Make sure community members are aware of the pre-reqs:
      - The Microsoft Learn path: [Discover the role of Python in Space Exploration](https://aka.ms/SpacewithPython-8). If they have not completed that path, an understanding of knowing how to code basic Python is recommended
      - Visual Studio Code installed
      - Python installed
      - Python extension in Visual Studio Code
2. **[Introduction to rocket launches](https://docs.microsoft.com/en-us/learn/modules/introduction-rocket-launch-nasa/)** 15-20 min
    - Resources for predicting weather
    - Analyze weather on launch day to determine safety
    - The basic machine learning process
    - The role of data in machine learning
    - Choose the right machine learning algorithm
    - How machine learning models learn
    - Walk through knowledge check questions
3. **[Data collection and manipulation](https://docs.microsoft.com/en-us/learn/modules/collect-manipulate-data-python-nasa/)** 30 min
    - Determine the rocket launch questions to ask
    - Explore the rocket launch data
    - Import Python libraries and rocket launch data
    - Clean weather data
    - Consider additional data
    - Walk through knowledge check questions
4. **Next steps** 2 min
    - Encourage attendees to complete the 1 remaining module in the learning path so they can get a trophy, and learn more about machine learning
5. **Closing** 5 min
    - Q&A
    - **Ask attendees to take the survey: https://aka.ms/Reactor/Survey**
      - Note: The event code is listed in this issue description and/or your Reactor contact can share that with the community members during the workshop
    - Share Resources
      - Foundations of data science Learn path: https://aka.ms/learn-reactor-foundations-ds
      - Get started with Machine Learning with an Azure Data Science Virtual Machine Learn path: https://aka.ms/learn-reactor-azureds-vm
      - Explain machine learning models with Azure Machine Learning Learn path: https://aka.ms/learn-reactor-learn-ml
      - Developer's Intro to Data Science video series: https://channel9.msdn.com/Series/Dev-Intro-to-Data-Science
    - Check if the Reactor contact has other information to share

**Comment by mebaumb** [comment]
All events published Dynamics - except Sao Paulo, waiting on localization for that one.

**Comment by mGalarnyk** [comment]
Just a reminder that I left my current job so I wont be giving either the Bangalore or US talks next week.

**Comment by meaghanlewis** [comment]
@mGalarnyk thanks for the reminder. Wishing you all the best in your new role!

**Comment by mGalarnyk** [comment]
Thank you!!! Hope all is well with you!
